internal/databases/postgres/queryRunner.go

```diff
@@ -385,3 +395,109 @@ func (queryRunner *PgQueryRunner) GetPhysicalSlotInfo(slotName string) (Physical
 func (queryRunner *PgQueryRunner) IsTablespaceMapExists() bool {
 	return queryRunner.Version >= 90600
 }
+
+// BuildCreateGreenplumRestorePoint formats a query to create a restore point for Greenplum
+func (queryRunner *PgQueryRunner) buildCreateGreenplumRestorePoint(restorePointName string) string {
```

Comment:
Better move the Greenplum-specific functions to the separate query runner.
You can introduce the GpQueryRunner and embed the existing PgQueryRunner into it.